### DISCUSSION TITLE
Generating New Config File During Airflow Upgrade

### DISCUSSION THREAD

**Discussion by SamiRehmanNostra** [apache/airflow#35788] [Q&A] [answered]
I am currently upgrading Apache Airflow from version 2.6.3 to 2.7.3 and have encountered an issue related to the configuration file. During the upgrade process, my existing airflow.cfg did not update automatically to reflect the new version's configurations.

I am seeking advice on the following:

Is there a standard procedure for generating a new airflow.cfg file when upgrading Airflow to ensure it includes all necessary updates and new settings?

If manual intervention is required, what are the best practices for comparing and updating the existing config file against the new version's default configurations?

Are there any tools or scripts recommended by the community for automating this process or simplifying the comparison between different versions of the configuration file?

Your insights and suggestions will be greatly appreciated, as I aim to ensure a smooth and error-free upgrade process.

**Comment by potiuk** [comment, marked as answer]
Read the advice about `airflow config list` command in https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html - it explains recommended production approach.

**Reply by SamiRehmanNostra** [reply]
Thanks mate, I don't know how I missed this :)
